rackcity/models/asset.py

```diff
@@ -23,6 +41,37 @@ def validate_hostname(value):
                               "it is not compliant with RFC 1034.")
 
 
+def validate_hostname_uniqueness(value, asset_id, change_plan, related_asset):
+    print(related_asset)
+    assets, assets_cp = get_assets_for_cp(change_plan.id)
+    matching_assets = assets_cp.filter(hostname=value)
+    if len(matching_assets) > 0 and matching_assets[0].id != asset_id:
+        raise ValidationError("'" + value + "'is not a unique hostname. \
+            A existing asset on this change plan exists with this hostname.")
+
+    matching_assets = assets.filter(hostname=value)
+    if len(matching_assets) > 0 and related_asset and matching_assets[0].id != related_asset.id :
+        raise ValidationError("'" + value + "'is not a unique hostname. \
+            A existing asset exists with this hostname.")
+
```

Comment:
custom hostname uniqueness validation -> this is only needed on AssetCP since you can have same hostnames across diff change plans